frontend/src/components/AllQueues.js

```diff
@@ -80,35 +88,48 @@ function QueueTab(props) {
   )
 }
 
-function AllQueues() {
+const AllQueues = observer(() => {
+  const queueStore = useContext(QueueStoreContext)
+  const queueSize = Object.keys(queueStore.queues).length
+
+  useEffect(() => {
+    for (let i = 1; i <= queueSize; i++) queueStore.updateQueue(i)
```

Comment:
so the way you have this set up is that every time you change tabs all the queues get called.  You could definitely have some logic here that doesnt do that.  I have some thoughts on how to do it, so if you dont know where to start or what to do or whatever let me know

Reply:
The database calls are in parallel, so I didn't worry about it as a first cut.  I think the way to optimize the calls to the back end is to create a hash for the queue data, and use that hash in useEffect so it knows if there has to be a back end call.

Reply:
But I'm happy to discuss it

Reply:
Oh I see.  I don't have to loop over all the queues, since they aren't being displayed.

Reply:
One reason I updated the whole set is because the tabs reflect the queue size and wait time. If only one were updated the queue lengths would be stale in a hard to understand way.

Reply:
I think its fine to get them all up front.  My only point is that there is no need to call getting all the queues once again every time you switch tabs.  Or is there?  Can you change the wait time or queue size from the queues view?